### PR TITLE
First pass at reworking injection/extraction API.

### PR DESCRIPTION
**Pull request by Bubblyworld**
See #158. Implements the changes described there, namely removing `jspm inject` and `jspm extract` and replacing them with the `-m/--map`, `-o/--output` flags on `jspm install` and `jspm inject` instead. This is still WIP, as I haven't updated docs or tested the changes properly yet, and extracting an import map from an input HTML file is unimplemented (might need to expose an API for it in the generator first).

Other than that, playing around with this seems to work, and I've migrated some simple tests to use the new flags (which pass).

### CODE REVIEW

**Comment by Bubblyworld** [comment]
Also fixes the issue with inject not writing `env` fields, but that's minor.

**Comment by Bubblyworld** [comment]
...and also haven't implemented the changes to `jspm link <blah>` inputs, currently it's still taking module specifiers as it was doing before.

File: src/link.ts [hunk 1]
```diff
@@ -21,17 +34,23 @@ export default async function link(packages: string[], flags: Flags) {
...
   if (!resolvedModules.length)
-    throw new JspmError('Trace install requires at least one module to trace.')
+    throw new JspmError('Linking requires at least one module to trace.')
   await generator.traceInstall(resolvedModules.map(p => p.target))
+
+  // If the input and output maps are the same, we behave in an additive way
+  // and trace all top-level pins to the output file. Otherwise, we behave as
+  // an extraction and only trace the provided modules to the output file.
+  let outputMap = generator.getMap()
+  if (inputMapPath !== outputMapPath)
```
Comment (guybedford) [inline src/link.ts:47]: What if we made this:

_The input map and output map are different **and there are no args**_.

Comment (guybedford) [reply src/link.ts:47]: Then we might not even need to have the `jspm link x.html` working at all. And can instead just use `jspm link -m x.html`.

**Pull request merged**